src/Optimizer/Transformer/PreloadHeroImage.php

```diff
@@ -194,24 +194,41 @@ public function transform(Document $document, ErrorCollection $errors)
      */
     private function findHeroImages(Document $document)
     {
-        $heroImageCandidate = null;
-        $heroImages         = [];
-        $node               = $document->body;
+        $heroImages                = [];
+        $heroImageCandidates       = [];
+        $heroImageFallbacks        = [];
+        $previousHeroImageFallback = null;
+        $node                      = $document->body;
 
         while ($node !== null) {
             if (! $node instanceof Element) {
                 $node = $this->nextNode($node);
                 continue;
             }
 
-            $heroImage = $this->detectImageWithDataHero($node);
+            $heroImage = $this->detectImageWithAttribute($node, Attribute::DATA_HERO);
             if ($heroImage) {
                 $heroImages[] = $heroImage;
+            } elseif (count($heroImageCandidates) < self::DATA_HERO_MAX) {
+                $heroImageCandidate = $this->detectImageWithAttribute($node, Attribute::DATA_HERO_CANDIDATE);
+                if ($heroImageCandidate) {
+                    $heroImageCandidates[] = $heroImageCandidate;
+                } elseif (count($heroImageFallbacks) < self::DATA_HERO_MAX) {
+                    $heroImageFallback = $this->detectPossibleHeroImageFallbacks($node);
+                    if ($heroImageFallback) {
+                        // Ensure we don't flag the same image twice. This can happen for placeholder images, which are
+                        // flagged on their own and as their parent's placeholder.
+                        if (
+                            ! $previousHeroImageFallback
+                            || $heroImageFallback->getAmpImg() !== $previousHeroImageFallback->getAmpImg()
+                        ) {
```

Comment:
Should these two conditions be combined? 

```suggestion
                    if (
                        $heroImageFallback &&
                        // Ensure we don't flag the same image twice. This can happen for placeholder images, which are
                        // flagged on their own and as their parent's placeholder.
                        (
                            ! $previousHeroImageFallback
                            || $heroImageFallback->getAmpImg() !== $previousHeroImageFallback->getAmpImg()
                        )
                    ) {
```